### PR TITLE
Includes the option to provide a durable ec2 client to ease testing

### PR DESCRIPTION
**Pull request by apozsuse**
Includes the option to provide a boto3.client class that will be reused in the instance.
The intention is to make tests simpler and improve test coverage.

### CODE REVIEW

**Review by smarlowucf — APPROVED** [review]
LGTM. It'd also be nice for the client to be a lazy attribute which would greatly simplify code. Instead of calling _connect everywhere it would just be self.ec2_client and that would get a new session/client if necessary. But that's a much bigger change.

**Comment by rjschwei** [comment]
> LGTM. It'd also be nice for the client to be a lazy attribute which would greatly simplify code. Instead of calling _connect everywhere it would just be self.ec2_client and that would get a new session/client if necessary. But that's a much bigger change.

The problem is that one does not know when the client times out. So if you set up lazy behavior, i.e. store the client once one still has to test whether the connection is actually alive or not. And that test would have to happen every time. So the choice is to have code that caches the client and then tests every time it is used whether the connection is still alive or to simply connect every time the client is used. I do not think the toolkit has changed in this respect. Such behavior should be the responsibility of the toolkit.

File: lib/ec2imgutils/ec2imgutils.py [hunk 1]
```diff
@@ -41,10 +46,16 @@ def __init__(self, log_level=logging.INFO, log_callback=None):
...
 
+        self.durable_ec2_client = ec2_client
+
     # ---------------------------------------------------------------------
     def _connect(self):
         """Connect to EC2"""
 
+        # Allow dependency injection for easier testing
+        if self.durable_ec2_client:
+            return self.durable_ec2_client
```
Comment (rjschwei) [inline lib/ec2imgutils/ec2imgutils.py:57]: But the client is not durable by definition. Neither boto3 nor botocore manage the connect. They basically just pass the connection to the API through. This means the connection **will** time out. As such at some point the durable client returned will be useless and broken.

**Comment by rjschwei** [comment]
This introduces breakage because there is no "durable client". There is nothing in the toolkit that manages the refresh of the connection and the connection will timeout. This is not a workable solution to whatever problem you are trying to solve. I do not claim that I understand the problem you are trying to solve.

File: lib/ec2imgutils/ec2imgutils.py (hunk 1 above)
Comment (smarlowucf) [reply lib/ec2imgutils/ec2imgutils.py:57]: Afaik there are ways to self manage sessions and set "TTL". That would provide a way to know when session is expiring and do a refresh. But that falls into a more expanded topic that is beyond the scope of this PR at the moment.

**Pull request closed without merging**